src/commands.rs

```diff
@@ -81,16 +90,23 @@ impl CommonOptions {
         config: &Config,
         retry: Option<Retry>,
     ) -> Result<FileSystemClient, ClientError> {
-        let client = match FileSystemClient::try_new_with_config(self.registry.as_deref(), config)?
-        {
+        let client = match FileSystemClient::try_new_with_config(
+            self.registry.as_deref(),
+            config,
+            self.auth_token(config)?.map(Secret::from),
```

Comment:
Looks like `auth_token` already returns a `Secret`? :thinking: 

```suggestion
            self.auth_token(config)?,
```